doc/reference/reference_lua/msgpack.rst

```diff
@@ -657,65 +658,100 @@ and :ref:`YAML <yaml-cfg>`.
-        msgpack.is_object({}) -- returns false
+
+..  _msgpack-module-api-reference-objects:
+
+Related objects
+~~~~~~~~~~~~~~~
 
 ..  _msgpack-object-methods:
 
-..  class:: msgpack_object
+msgpack_object
+**************
 
-    A MsgPack object can be passed to the MsgPack encoder with the same effect as passing the original Lua object:
+..  class:: msgpack_object
 
-    ..  code-block:: lua
+    A MsgPack object that stores arbitrary MsgPack data.
+    To create a MsgPack object from a Lua object or string, use the following methods:
 
-        local msgpack = require('msgpack')
-        local mp = msgpack.object(123)
-        msgpack.object({mp, mp}):decode()         -- returns {123, 123}
-        msgpack.decode(msgpack.encode({mp, mp}))  -- returns {123, 123}
+    *   :ref:`msgpack.object <msgpack-object>`
+    *   :ref:`msgpack.object_from_raw <msgpack-object-from-raw>`
 
-    In particular, this means that if a MsgPack object stores an array, it can be inserted into a database space:
+    If a MsgPack object stores an array, it can be inserted into a database space:
 
     ..  code-block:: lua
 
-        box.space.my_space:insert(msgpack.object({1, 2, 3}))
-
-    The MsgPack object has the following methods:
+        box.space.bands:insert(msgpack.object({1, 'The Beatles', 1960}))
 
     ..  method:: decode()
 
-        Since version 2.10.0.
+        **Since:** :doc:`2.10.0 </release/2.10.0>`
 
         Decode MsgPack data in the MsgPack object.
 
         :return: a Lua object
 
         :rtype: Lua object
 
+        **Example**
+
+        ..  literalinclude:: /code_snippets/test/msgpack/msgpack_object_test.lua
+            :language: lua
+            :lines: 1-2,4-5,19-22
+            :dedent:
+
     ..  method:: iterator()
 
-        Since version 2.10.0.
+        **Since:** :doc:`2.10.0 </release/2.10.0>`
 
         Create an iterator over the MsgPack data.
 
-        A MsgPack iterator object has its own :ref:`set of methods <msgpack-object-iterator-methods>`.
-
-        :return: an iterator object over the MsgPack data
+        :return: an :ref:`iterator object <msgpack-object-iterator-methods>` over the MsgPack data
 
         :rtype: userdata
 
+    ..  method:: get(key)
+
+        **Since:** :doc:`2.11.0 </release/2.11.0>`
+
+        Get an element of the MsgPack array by the specified index key.
+        You can also use the indexed notation (``[]``) to get an array element.
+
+        :param number/string key: the index key used to get the array element, which might be one of the following:
+
+                    *   if a MsgPack object is an array, ``key`` is the integer value (start with 1) that specifies the element index.
+                    *   if a MsgPack object is an associative array, ``key`` is the string value that specifies the element key.
+
+        :return: an element of the MsgPack array.
+                 If the specified key is missing in the array, `get` returns ``nil``.
```

Comment:
May be worth mentioning that this function raises an error in case the value stored in the object isn't a MsgPack array or map.